File: xmatch_web/src/main/java/com/zsq/service/UserService.java
package com.zsq.service;

import com.zsq.model.*;
import java.util.Map;

/**
 * @author CoderQiang
 * Created by coderqiang on 2017/11/4.
 */

public interface UserService {
    /**
     * 添加用户
     * @param user
     * @return
     */
    public Map<String,Object> addUser(User user);

    public int addUserAvator(long userId,String url);

    /**
     * 更新用户
     * @param user
     * @return
     */
    public int updateUser(User user);

    public User getUser(long userId);

    public User getUserByStuNo(String stuNo);

    /**
     * 登陆
     */
    public Map<String, Object> userLogin(String stuNo, String passwd);

}

File: xmatch_android/app/src/main/java/com/example/coderqiang/xmatch_android/api/service/UserService.kt
package com.example.coderqiang.xmatch_android.api.service

import com.example.coderqiang.xmatch_android.dto.BaseMessage
import com.example.coderqiang.xmatch_android.dto.ObjectMessage
import com.example.coderqiang.xmatch_android.model.DepManager
import com.example.coderqiang.xmatch_android.model.DepMember
import com.example.coderqiang.xmatch_android.model.Department
import com.example.coderqiang.xmatch_android.model.User
import retrofit2.Call
import retrofit2.http.*

/**
 * Created by coderqiang on 2017/11/11.
 */
interface UserService {

    @POST("api/user/login")
    fun userLogin(@Query("stuNo") depManagerAccount:String,
                  @Query("passwd") password:String): Call<ObjectMessage<User>>

    @POST("api/user/add")
    fun addUser(@Body user:User):Call<ObjectMessage<Integer>>

    @POST("api/user/getDeps")
    fun getUserDeps(@Query ("userId") userId:Long):Call<ObjectMessage<List<Department>>>

    @POST("/api/depMemberManage/add")
    fun addDepApply(@Body depMember: DepMember):Call<BaseMessage>

//    @GET("api/user/get")
//    fun getDepManagerById(@Path("id")depManagerId:Long): Call<ObjectMessage<DepManager>>

}